docs/reference/api/drivers/AnalogIn.md

```diff
@@ -2,6 +2,12 @@
 
 Use the AnalogIn API to read an external voltage applied to an analog input pin. `AnalogIn()` reads the voltage as a fraction of the system voltage. The value is a floating point from `0.0`(VSS) to `1.0`(VCC). For example, if you have a 3.3V system and the applied voltage is 1.65V, then `AnalogIn()` reads `0.5` as the value.
 
+ One of the most common types of analog to digital converters used in microcontrollers today is called the successive-approximation ADC. With successive-approximation, the analog input to the ADC is essentially guessed in a binary search-like fashion by the ADC hardware, which consists of a voltage comparator, a DAC, and a successive approximation register (SAR), which will hold the result of the conversion. For a 12-bit ADC, after 12 clock cycles, the value in the SAR will converge on the value of the analog input, thus completing the voltage sampling process. 
```

Comment:
Overall the document looks great, particularly the resolution section. It would be nicer if we can make the DAC description even more shorter. The focus here is to explain the API and not necessarily how the DAC works.